clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1505,9 +1506,27 @@ public void resume(Collection<TopicPartition> partitions) {
      */
     @Override
     public void close() {
+        close(DEFAULT_CLOSE_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Tries to close the consumer cleanly within the specified timeout. This method waits up to
+     * <code>timeout</code> for the consumer to complete pending commits and leave the group.
+     * If the consumer is unable to complete the requests before the timeout expires, the consumer
+     * is force closed.
```

Comment:
I have more of a question than a comment. In the KIP discussion thread, a question was asked about happens if the consumer coordinator is not known when the `consumer.close()`, is called. It appears to me that if auto commit is enabled, then we wait upto the timeout limit to find the coordinator. In other cases, we close immediately. 

Is this correct? If so, would it be worth mentioning this nuance in the javadoc?